homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,263 @@
+"""samsungctl and samsungtvws bridge classes."""
+import os
+
+import requests
+from requests import RequestException
+from samsungctl import Remote
+from samsungctl.exceptions import AccessDenied, ConnectionClosed, UnhandledResponse
+from samsungtvws import SamsungTVWS
+from websocket import WebSocketException
+
+from homeassistant.const import STATE_OFF, STATE_ON
+
+from .const import (
+    LOGGER,
+    RESULT_AUTH_MISSING,
+    RESULT_NOT_SUCCESSFUL,
+    RESULT_NOT_SUPPORTED,
+    RESULT_SUCCESS,
+)
+
+
+class SamsungTVBridge:
+    """The Base Bridge class."""
+
+    @staticmethod
+    def get_bridge(config):
+        """Get Bridge instance."""
+        if config:
+            if config["method"] == "legacy":
+                return SamsungTVLegacyBridge(config)
+            if config["method"] == "websocket":
+                return SamsungTVWSBridge(None, config)
+        return None
+
+    def __init__(self, config):
+        """Initialize Bridge."""
+        self.port = None
+        self.method = None
+        self.token_file = None
+        self.config = None
+        self._remote = None
+        self._callback = None
+
+    def register_reauth_callback(self, func):
+        """Register a callback function."""
+        self._callback = func
+
+    def try_connect(self, host, port):
+        """Try to connect to the TV."""
+
```

Comment:
Don't we need a no-op `pass` here?

Reply:
And you should save `host` and `port` at `__init__`.

Reply:
Not sure `pass` is really needed (the comment should be enough)
I was thinking of making the `try_connect*` methods as static.